components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -3556,6 +3558,7 @@ public static AuthenticatedUser createAuthenticatedUser(String username, String
             } else {
                 authenticatedUser.setFederatedIdPName(OAuth2Util.getFederatedIdPFromDomain(userStoreDomain));
             }
+            authenticatedUser.setUserId(getUserIdOfFederatedUser(username, tenantDomain, idpName));
```

Comment:
will the scenario work as expected for a local user? don't we need to set the user ID for a local user as well?

Reply:
For local users, we have a different way of resolving the user id and it's handled separately.